File: client/Netmodule/src/Messages/MessageCodecFactory.cpp
#include "MessageCodec.h"
#include "Message.h"

namespace Netmodule{
/* --------------------------------------------------------------------------------------------------
 *      MessageEncoder
 * -------------------------------------------------------------------------------------------------- */  
    void MessageEncoder::encode(sp<EIoSession>& session, sp<EObject>& message, sp<EProtocolEncoderOutput>& out) THROWS(EException){
        Message *msg = static_cast<Message*>(message.dismiss());
        out->write(msg->getEncodedMessage());
    }
    void MessageEncoder::dispose(sp<EIoSession>& session) THROWS(EException) {}
/* --------------------------------------------------------------------------------------------------
 *      Messagedecoder
 * -------------------------------------------------------------------------------------------------- */  
    void MessageDecoder::decode(sp<EIoSession>& session, EIoBuffer* in, sp<EProtocolDecoderOutput>& out) THROWS(EException){
        out->write(Message::decodeBytesToMessage(in));
    }
    void MessageDecoder::finishDecode(sp<EIoSession>& session, sp<EProtocolDecoderOutput>& out) THROWS(EException) {}
    void MessageDecoder::dispose(sp<EIoSession>& session) THROWS(EException) {}
/* --------------------------------------------------------------------------------------------------
 *      MessageCodecFactory
 * -------------------------------------------------------------------------------------------------- */  
    MessageCodecFactory::MessageCodecFactory() {
        this->encoder = new MessageEncoder();
        this->decoder = new MessageDecoder();
    }

    EProtocolEncoder* MessageCodecFactory::getEncoder(sp<EIoSession>& session) THROWS(EException){
        return encoder;
    }

    EProtocolDecoder* MessageCodecFactory::getDecoder(sp<EIoSession>& session) THROWS(EException){
        return decoder;
    }

}

File: server/src/main/java/CoolChat/Messages/MessageCodecFactory.java
package CoolChat.Messages;

import org.apache.mina.core.session.IoSession;
import org.apache.mina.filter.codec.ProtocolCodecFactory;
import org.apache.mina.filter.codec.ProtocolDecoder;
import org.apache.mina.filter.codec.ProtocolEncoder;

public class MessageCodecFactory implements ProtocolCodecFactory{
    private ProtocolEncoder encoder;
    private ProtocolDecoder decoder;

    public MessageCodecFactory(){
        encoder = new MessageEncoder();
        decoder = new MessageDecoder();
    }

    public ProtocolEncoder getEncoder(IoSession session)throws Exception{
        return encoder;
    }

    public ProtocolDecoder getDecoder(IoSession session)throws Exception{
        return decoder;
    }
}